### ISSUE TITLE
[java] New rule: OnDemandImport

### ISSUE THREAD

**Issue by zbynek** [pmd/pmd#7047]
**Proposed Rule Name:** OnDemandImport

**Proposed Category:** Best Practices

**Description:**
On-demand imports (also called star imports) should be use sparingly. By default, on-demand imports should be allowed only for static methods of supported test frameworks. These exclusions should be configurable by properties.

Related Checkstyle rule: https://checkstyle.sourceforge.io/checks/imports/avoidstarimport.html

**Code Sample:** 
```java
import static org.junit.jupiter.api.Assertions.* // this is allowed by default
import static java.lang.Math.*; // violation
import java.util.*; // violation
```

**Possible Properties:**

*   `allowStaticImportsFrom` -- set to `*` to allow all static imports, or a list of packages (default `org.junit.jupiter.api,org.junit,org.testng`)
*  `allowTypeImportsFrom` -- set to `*` to allow all type imports, or a list of packages (default empty)

**Comment by suhanrain** [comment]
Hi, I'd like to work on this. I plan to follow the existing Java import-rule patterns and cover the default exclusions and both configurable properties with tests. Could you assign it to me?

**Comment by zbynek** [comment]
@suhanrain thanks for offering to work on this -- I think there's no need to formally assign it, just stating your intent like you did should be enough.
